core/src/main/java/org/pentaho/platform/web/http/api/resources/GenericFileResource.java

```diff
@@ -111,6 +114,8 @@ public IGenericFileTree getFileSubtree( @NonNull @PathParam( "path" ) String bas
       throw new WebApplicationException( e, Response.Status.FORBIDDEN );
     } catch ( InvalidPathException | IllegalArgumentException e ) {
       throw new WebApplicationException( e, Response.Status.BAD_REQUEST );
+    } catch ( NotFoundException e ) {
+      throw new WebApplicationException( e, Response.Status.NOT_FOUND );
```

Comment:
Was missing this mapping, resulting in a 500 when base path was not found, not a folder or invalid connection folder. The UI was already properly reacting correctly (it's not successful whether 404 or 500), but this is the status code that should be being returned in this situation all along.